qiskit_nature/second_q/mappers/mode_based_mapper.py

```diff
@@ -0,0 +1,143 @@
+# This code is part of a Qiskit project.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Mode Based Mapper."""
+
+from __future__ import annotations
+
+from typing import Union
+from abc import abstractmethod
+
+import numpy as np
+from qiskit.quantum_info.operators import Pauli, PauliList, SparsePauliOp
+
+from qiskit_nature import QiskitNatureError
+from qiskit_nature.second_q.operators import SparseLabelOp
+from qiskit_nature.second_q.mappers.qubit_mapper import QubitMapper
+
+# Types that can be data for a SparsePauliOp
+PauliType = Union[PauliList, SparsePauliOp, Pauli, list, str]
+
+
+class ModeBasedMapper(QubitMapper):
+    """Mapper from ``SparseLabelOp`` to a qubit operator using a Pauli table."""
+
+    def _map_single(
+        self, second_q_op: SparseLabelOp, *, register_length: int | None = None
+    ) -> SparsePauliOp:
+        return self.mode_based_mapping(second_q_op, register_length=register_length)
+
+    @abstractmethod
+    def pauli_table(self, register_length: int) -> list[tuple[PauliType, PauliType]]:
+        r"""Generates a Pauli-lookup table mapping from modes to Pauli operators or pairs of Pauli
+        operators.
+
+        This table is a list of tuples :math:`(P, Q)` of two Pauli operators, corresponding to the
+        real part :math:`P` and imaginary part :math:`Q` for the respective mode index. These Pauli
+        operators are used to construct the creation and annihilation operators
+        :math:`(P \pm i Q)/2`.
+
+        The generated table is processed by :meth:`.QubitMapper.sparse_pauli_operators`.
```

Comment:
```suggestion
        The generated table is processed by :meth:`.sparse_pauli_operators`.
```
This is actually part of `ModeBasedMapper` now, but since this is inside the same class, the link should resolve like this as well.

Reply:
Thanks for spotting this.